tests/Microsoft.Identity.Test.Unit/UtilTests/HttpHeaderSantizerTests.cs

```diff
@@ -0,0 +1,39 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Text;
+using Microsoft.Identity.Client.TelemetryCore.Http;
+using Microsoft.Identity.Test.Common;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Microsoft.Identity.Test.Unit.UtilTests
+{
+    [TestClass]
+    public class HttpHeaderSantizerTests
+    {
+        [TestMethod]
+        // For https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/1881
+        public void SantizerAllowsStrangeErrorCodesToBeSentAsHeaders()
+        {
+            // this is a real B2C error code :(
+            string strangeErrorCode = "An error occured, error: access_denied, error_description: AADB2C90118: The user has forgotten their password. "
+                + Environment.NewLine +
+                "Correlation Id: " + Guid.NewGuid()
+                + Environment.NewLine +
+                "Timestamp " + DateTime.UtcNow
+                + Environment.NewLine +
+                "An error occured, error: access_denied, error_description: AADB2C90118: The user has forgotten their password." +
+                "Correlation Id: " + Guid.NewGuid()
+                + Environment.NewLine +
+                "Timestamp " + DateTime.UtcNow;
+            HttpRequestMessage httpRequest = new HttpRequestMessage();
+
+            // newline chars must be followed by spaces
```

Comment:
>newline chars must be followed by spaces [](start = 15, length = 40)

seems weird that this throws an exception...

Reply:
Apparently the rule this breaks is "newlines must be followed by a space"

Reply:
probably that is what the solution should have indicated? this could have been part of the doc for the sanitizer class?

---
In reply to: [526875315](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/2209#discussion_r526875315) [](ancestors = 526875315)